### PR TITLE
ON-27063 # Changed verifyJWT to call API instance of using cognito

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: OneBlink.SDK/FormsAppsClient.cs [hunk 1]
```diff
@@ -22,17 +23,14 @@ public FormsAppsClient(string accessKey, string secretKey, TenantName tenantName
...
                 token = token.Split(' ')[1];
             }
-            JwtSecurityToken jwt = new JwtSecurityToken(token);
-            string iss = this.oneBlinkApiClient.tenant.jwtIssuer;
-            JsonWebKey jwk = await Token.GetJsonWebKey(
-                iss: iss,
-                kid: jwt.Header.Kid
-            );
-            string verifiedToken = Token.VerifyJWT(token, jwk, iss);
+            string verifiedToken = Token.DecodeJWT(token);
```
Comment (kizaonline) [inline OneBlink.SDK/FormsAppsClient.cs:33]:
```suggestion
           RawJWTPayload rawJWTPayload = JsonConvert.DeserializeObject<RawJWTPayload>(token);
            if (String.IsNullOrEmpty(rawJWTPayload.sub))
            {
                return null;
            }
```

Comment (mymattcarroll) [reply OneBlink.SDK/FormsAppsClient.cs:33]: `token` is a JWT string, not a JSON string.

Comment (kizaonline) [reply OneBlink.SDK/FormsAppsClient.cs:33]: I was trolling :)

File: OneBlink.SDK/Token.cs [hunk 2]
```diff
@@ -65,7 +65,16 @@ internal static string GenerateJSONWebToken(string accessKey, string secretKey,
...
-        internal static string VerifyJWT(string token, JsonWebKey jwk, string iss) {
+
+        internal static string DecodeJWT(string token)
+        {
+            JwtSecurityTokenHandler handler = new JwtSecurityTokenHandler();
+            JwtSecurityToken validatedJwt = handler.ReadJwtToken(token);
+            return validatedJwt.Payload.SerializeToJson();
+        }
+
+        internal static string VerifyJWT(string token, JsonWebKey jwk, string iss)
```
Comment (kizaonline) [inline OneBlink.SDK/Token.cs:76]: This can be removed now and any now unused deps

File: OneBlink.SDK.Tests/FormsAppsClientTests.cs [hunk 3]
```diff
@@ -46,22 +46,13 @@ public void throws_error_if_keys_empty()
             Assert.Throws<ArgumentException>(() => new FormsAppsClient("", ""));
         }
 
-        [Fact]
```
Comment (kizaonline) [inline OneBlink.SDK.Tests/FormsAppsClientTests.cs:49]: Why has this test been removed? Will these tests pass after the migration?

File: OneBlink.SDK/FormsAppsClient.cs [hunk 4]
```diff
@@ -22,17 +22,13 @@ public FormsAppsClient(string accessKey, string secretKey, TenantName tenantName
...
```
Comment (kizaonline) [inline OneBlink.SDK/FormsAppsClient.cs:31]:
```suggestion
            string decodedToken = Token.DecodeJWT(token);
```

File: OneBlink.SDK.Tests/FormsAppsClientTests.cs (hunk 3 above)
Comment (mymattcarroll) [reply OneBlink.SDK.Tests/FormsAppsClientTests.cs:49]: Code no longer throws this error and yes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -9,6 +9,7 @@
 ### Changed
 
 -   Origin for test tenant's
+-   `FormsAppsClient.VerifyJWT()` to call API instance of using cognito
```
Comment (kizaonline) [inline CHANGELOG.md:12]:
```suggestion
-   `FormsAppsClient.VerifyJWT()` to call API instance instead of using cognito
```

**Review by kizaonline — APPROVED** [review]

**Pull request merged**